### ISSUE TITLE
Possible bug or user error? Echoes look suspitiously similar after fmriprep.

### ISSUE THREAD

**Issue by dangom** [poldracklab/fmriprep#1131]
Hi guys,

we have an issue here but it's possible that we are the ones doing something wrong.
In one of our experiments we acquire a 3-echo dataset and preprocess it with fmriprep 1.0.13 running the following command:

> FMRIPrep command: /home/mrphys/zahfaz/anaconda3/bin/fmriprep --participant-label sub-07 --low-mem --use-syn-sdc --ignore fieldmaps slicetiming --use-aroma --aroma-melodic-dimensionality 100 --no-submm-recon --output-space T1w template -w /data/fmripreptest-task-sub-07 --nthreads 12 --fs-license /home/mrphys/zahfaz/usr/local/freesurfer/license.txt --fs-no-reconall /project/3015046.02/task-experiment-datasets /project/3015046.02/task-derivatives/temp participant

For completion, here's how the func directory looks like:

```
$ tree
.
|-- sub-07_ses-01_task-stroop_acq-mb_bold.json
|-- sub-07_ses-01_task-stroop_acq-mb_bold.nii.gz
|-- sub-07_ses-01_task-stroop_acq-mbme_echo-01_bold.json
|-- sub-07_ses-01_task-stroop_acq-mbme_echo-01_bold.nii.gz
|-- sub-07_ses-01_task-stroop_acq-mbme_echo-02_bold.json
|-- sub-07_ses-01_task-stroop_acq-mbme_echo-02_bold.nii.gz
|-- sub-07_ses-01_task-stroop_acq-mbme_echo-03_bold.json
`-- sub-07_ses-01_task-stroop_acq-mbme_echo-03_bold.nii.gz
```
BIDS-Validator says things are OK except for a missing 'TaskName', but I wouldn't expect that to be relevant to the issue at hand.

Our problem is that after processing, all of the three echoes look extremely alike, as you can see in the shots below:

Echo 1 **raw**
![echo-01](https://user-images.githubusercontent.com/7881842/40056632-478db956-584c-11e8-9704-c7ef494af72d.png)
Echo 1 **fmriprep**
![echo-01_preproc](https://user-images.githubusercontent.com/7881842/40056637-4a3d3b86-584c-11e8-8072-5c1fa48a286d.png)
Echo 2 **raw**
![echo-02](https://user-images.githubusercontent.com/7881842/40056639-4bb9e608-584c-11e8-8589-e852a72c8b0f.png)
Echo 2 **fmriprep**
![echo-02_preproc](https://user-images.githubusercontent.com/7881842/40056649-4da76850-584c-11e8-84f5-d1ac4862438a.png)
Echo 3 **raw**
![echo-03](https://user-images.githubusercontent.com/7881842/40056651-4f038b20-584c-11e8-8998-d27e7e76fbd6.png)
Echo 3 **fmriprep**
![echo-03_preproc](https://user-images.githubusercontent.com/7881842/40056656-50a6ff2a-584c-11e8-88d9-15a7375724b6.png)

Apologize for the suboptimal contrast windows used for the raw datasets. These are screenshots taken from within FSLEYES.
In any case, the preprocessed echoes "look" identical despite containing different information. It also looks like the distortion correction was applied independently in each of them (shouldn't we have the same warp applied to all images? To a good approx. the distortion should be the same for all echoes).

Could it be that the first echo is being cloned somewhere along the pipeline and overwriting the other echoes?

Curious to hear your suggestions / interpretations. If you need any more info, let me know and I'll be happy to provide it.

**Comment by emdupre** [comment]
Just to jump in&mdash; if you'd like the same coregistration to be applied across multi-echo data sets, I'd recommend throwing the `--t2s-coreg` flag, which will generate a T2* map from the echos and then use that as the coregistration target. 

As to whether or not specifically the distortion should be the same across all echos, I'm not sure that's the case, and would advocate that the distortion correction is actually calculated from the T2* map rather than from any of the individual echos. This was something @oesteban and I were discussing in #1031

**Comment by oesteban** [comment]
Hi @dangom, you are right, this is very suspicious. I'll check we are not writing the same image three times. Since you are actively using the multi-echo processing it is probably a good time to make a revision of the ME pipeline and #1031. Can we count on your expert eyes to help?

**Comment by emdupre** [comment]
@dangom `--t2s-coreg` should only apply to the multi-echo run. If this is not your experience, please let us know !

@oesteban I just compared ds210 outputs and confirmed that derivatives of echo 1 and 2 for ds210 are the same. This is not the case with data I processed myself for fmriprep v 1.0.9, so it likely arose from a subsequent refactor.

**Comment by oesteban** [comment]
Thanks a lot for checking this. I'll try to find out where this is
happening.

**Comment by oesteban** [comment]
Effectively, this early node

https://github.com/poldracklab/fmriprep/blob/e5d1ad7d311e90b8dc14c26176d046d3060200bf/fmriprep/workflows/bold/base.py#L349

has just the first echo as input.

**Comment by emdupre** [comment]
Ok, I think (before `boldbuffer` was added as a field) we were passing just the first echo to STC since that metadata should be consistent across echos. Now that you've streamlined that part of the workflow, we're missing that `inputnode.bold_file` is actually an iterable. I can look into patching this.

**Comment by oesteban** [comment]
If you can get this off my plate, I'd really, deeply appreciate it :)

**Comment by oesteban** [comment]
Fixed via #1134. @dangom, could you please double check that we fixed the problem indeed?

**Comment by dangom** [comment]
Fixed. Thank you guys!
